setup.c

```diff
@@ -1368,9 +1374,17 @@ const char *setup_git_directory_gently(int *nongit_ok)
 		break;
 	case GIT_DIR_INVALID_OWNERSHIP:
 		if (!nongit_ok) {
+			struct strbuf prequoted = STRBUF_INIT;
 			struct strbuf quoted = STRBUF_INIT;
 
-			sq_quote_buf_pretty(&quoted, dir.buf);
+#ifdef __MINGW32__
+			if (dir.buf[0] == '/')
+				strbuf_addstr(&prequoted, "%(prefix)/");
```

Comment:
_If_ this turns out to be the necessary solution we need to prepend `%(preffix)/` to the unquoted as well as the quoted value. GitHub Desktop will use the unquoted value since we won't be passing it via a shell first

Reply:
We do need to add the `%(prefix)/`. Without it, Git collapses `//` to `/` as in this debugger output:

```
Thread 1 hit Breakpoint 1, safe_directory_cb (key=0x204a56d6278 "safe.directory", value=0x204a56d0fd8 "//wsl.localhost/Ubuntu/home/stolee/_git/git", d=0xf674fff220) at setup.c:1045
1045                    const char *interpolated = NULL;
(gdb) n
1047                    if (!git_config_pathname(&interpolated, key, value) &&
(gdb)
1048                        !fspathcmp(data->path, interpolated ? interpolated : value))
(gdb) p interpolated
$1 = 0x204a56d6a88 "/wsl.localhost/Ubuntu/home/stolee/_git/git"
(gdb) p data->path
$2 = 0x204a56d2808 "//wsl.localhost/Ubuntu/home/stolee/_git/git"
```

Then, we _do_ add this to the quoted output because `prequoted` is the buffer being quoted in `sq_quote_buf_pretty()` (after appending `dir.buf` to `prequoted`).

Reply:
Should we be more stringent and require _two_ slashes before doing this?

Reply:
I was using this hunk as a precedent for a single slash:

https://github.com/git-for-windows/git/blob/3f8d16356f5bcb484eb25a886bb68dcc419c348c/path.c#L742-L747

Reply:
> Then, we _do_ add this to the quoted output because `prequoted` is the buffer being quoted in `sq_quote_buf_pretty()` (after appending `dir.buf` to `prequoted`).

Right, I'm sorry, I'm not being terribly clear, it's late. What I meant was that with this change (if I'm reading it correctly) the output from the warning for a WSL path will be something like

```
unsafe repository ('//wsl/something' is owned by someone else)
To add an exception for this directory, call:

			      git config --global --add safe.directory %(prefix)//wsl/something
```

I.e the first path within the first line parenthesis will not have `%(prefix)` prepended right? What I'm saying is that it should because GitHub Desktop will parse that first path and use it to add the config.

I'm suggesting we swap out `dir.buf` on line 1392 for `prequoted.buf` so that both instances of the path has the `%(prefix)` prepended to them,

Reply:
Ah, I get what you're saying now, @niik.

I'm not sure that adding `%(prefix)/` there is a good idea, because that's not the internal representation of the path. The `%(prefix)/` concept is only for parsing paths from config.